src/tools/debugger/FrontendConnector.java

```diff
@@ -209,36 +211,139 @@ public void sendActorHistory() {
       actorObjsToIds.put(a, e.getValue());
     }
 
-    JSONObjectBuilder msg = JsonSerializer.createMessageHistoryJson(messagesPerThread,
-        actorsToIds, actorObjsToIds);
+    JSONObjectBuilder msg = JsonSerializer.createMessageHistoryJson(
+        messagesPerThread, actorsToIds, actorObjsToIds);
 
     String m = msg.toString();
     log("[ACTORS] Message length: " + m.length());
     sender.send(m);
     log("[ACTORS] Message sent?");
     try {
       Thread.sleep(150000);
-    } catch (InterruptedException e1) { }
+    } catch (InterruptedException e1) {
+
+    }
     log("[ACTORS] Message sent waiting completed");
 
     sender.close();
   }
 
   public void requestBreakpoint(final boolean enabled, final URI sourceUri,
-      final int startLine, final int startColumn, final int charLength) {
+      final int startLine, final int startColumn, final int charLength,
+      final String role) {
     try {
       Breakpoint bp = breakpoints.getBreakpoint(sourceUri, startLine, startColumn, charLength);
       bp.setEnabled(enabled);
+
+      ObjectBuffer<ObjectBuffer<ObjectBuffer<EventualMessage>>> mm = ActorExecutionTrace.getAllProcessedMessages();
+      ObjectBuffer<ObjectBuffer<SFarReference>> aa = ActorExecutionTrace.getAllCreateActors();
+
+      Actor actor = null;
+      boolean receiver;
+      if (role.equals("receiver")) {
+        // TODO get receiver actor
+        log("Send breakpoint on receiver");
+        receiver = true;
+      } else {
+        // TODO get sender actor
+        log("Send breakpoint on sender");
+        receiver = false;
+      }
+
+      log("breakpoint coordinates " + startLine + " " + startColumn + " " + charLength);
+
+      Map<BreakpointId, Breakpoint> storeBreakpoints = breakpoints.getKnownBreakpoints();
+      BreakpointId bId = getBreakpointId(storeBreakpoints, sourceUri, startLine, startColumn, charLength);
+
+      //get holder class and method name from root node from coordinates
+      BreakpointLocation location = getBreakpointLocation(sourceUri, startLine, bId);
+      if (location != null) {
+        log("holder class: " + location.getHolderClass());
+        log("method name: " + location.getMethodName());
+      }
+
+      if (actor != null) {
+       actor.getLocalManager().setFileName(sourceUri);
+       ActorExecutionTrace.assignBreakpoint(bp, actor, location, receiver);
+      }
+
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
   }
 
+  private BreakpointId getBreakpointId(final Map<BreakpointId, Breakpoint> storeBreakpoints, final URI sourceUri,
```

Comment:
This seems like functionality that belongs into the `Breakpoints` class, especially since you needed to add a getter to get all known breakpoints. I'd prefer to remove the getter, and have this id lookup being in `Breakpoints`.